### PR TITLE
feat(Touchable, Buttons): Improve touchable and buttons accessibility

### PR DESCRIPTION
**Pull request by pladaria**
https://jira.tid.es/browse/WEB-1794

This PR adds the `onNavigate` and `role` props to:
- `Touchable`
- `ButtonPrimary`, `ButtonSecondary`, etc
- `ButtonLink`
- `TextLink`

There are other touchable-like components like `IconButton`, `Row`, `Card`, `Chip`... We can add these features when required but I think the scope of this PR is Ok for now

### CODE REVIEW

File: src/utils/browser.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const redirect = (url: string, external = false, loadOnTop = false): void => {
```
Comment (pladaria) [inline src/utils/browser.tsx:1]: moved to an external file to facilitate testing

**Review by Winde — APPROVED** [review]

File: src/__tests__/button-test.tsx [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
+import type {ThemeConfig} from '../theme';
+
+jest.mock('../utils/browser', () => ({
+    ...jest.requireActual('../utils/browser'),
+    redirect: jest.fn(),
+}));
+
+beforeEach(() => {
+    (redirectSpy as any).mockReset();
+});
```
Comment (marcoskolodny) [inline src/__tests__/button-test.tsx:21]: Out of curiosity: why is this needed?

File: src/button.tsx [hunk 3]
```diff
@@ -377,7 +384,7 @@ const Button = React.forwardRef<TouchableElement, ButtonProps & {type: ButtonTyp
             EndIcon: props.EndIcon,
         }),
         disabled: props.disabled || showSpinner || isFormSending,
-        role: 'button',
+        role: props.role,
```
Comment (marcoskolodny) [inline src/button.tsx:387]: Shouldn't we default to button in case role was not provided?

File: src/touchable.tsx [hunk 4]
```diff
@@ -52,56 +44,80 @@ interface CommonProps {
...
-export interface PropsMaybeTo extends CommonProps {
+
+interface PropsMaybeTo extends CommonProps {
     maybe: true;
     to?: string | Location;
     fullPageOnWebView?: boolean;
     replace?: boolean;
     href?: undefined;
     onPress?: undefined;
+    /** with "to", onNavigate will be executed in parallel to the navigation */
```
Comment (marcoskolodny) [inline src/touchable.tsx:88]: As far as I can see, it's not in parallel, right? You wait for the promise to be resolved, and after that you navigate

File: playroom/components/index.tsx [hunk 5]
```diff
@@ -30,7 +30,7 @@ import IconCode from '../icons/icon-code';
 import * as styles from '../preview-tools.css';
 import {CSSTransition} from 'react-transition-group';
 
-import type {IconProps} from '../../dist';
+import type {IconProps} from '../../src/utils/types';
 import type {ThemeConfig, ColorScheme, KnownSkinName} from '../../src';
```
Comment (atabel) [inline playroom/components/index.tsx:34]: why don't you import `IconProps` form `'../../src'` too?

File: src/__tests__/button-test.tsx [hunk 6]
```diff
@@ -44,18 +64,13 @@ test('<a> is rendered when using "to" prop', () => {
         </ThemeContextProvider>
     );
 
-    const anchor = screen.getByRole('button', {name: 'test'});
+    const anchor = screen.getByRole('link', {name: 'test'});
```
Comment (atabel) [inline src/__tests__/button-test.tsx:67]: I know this is an expected change, but it will probably break many tests in webapp

File: src/touchable.tsx
Comment (atabel) [inline src/touchable.tsx:1]: Maybe we could have a `TouchableProps<T>` generic type to reuse all the `onPress`/`to`/`href` + friend props in any touchable component (Buttons, IconButton, Cards, Rows, etc).

I've created a ticket: https://jira.tid.es/browse/WEB-1817

**Review by atabel — APPROVED** [review]
Please, create a ticket to add `onNavigate` prop to cards, rows, etc

File: src/button.tsx (hunk 3 above)
Comment (pladaria) [reply src/button.tsx:387]: the default is provided by the dom element (button for `<button>` and link for `<a>`)

File: src/touchable.tsx (hunk 4 above)
Comment (pladaria) [reply src/touchable.tsx:88]: what you comment is true in the "href", not for the "to".

The "to" prop will use a `<Link>` element which produces a client side navigation, so it is ok to perform the onNavigate actions in parallel (this was already happening in current implementation)

File: playroom/components/index.tsx (hunk 5 above)
Comment (pladaria) [reply playroom/components/index.tsx:34]: ok, will do

**Comment by pladaria** [comment]
> Please, create a ticket to add `onNavigate` prop to cards, rows, etc

https://jira.tid.es/browse/WEB-1818
